l10n_th_base_location/models/res_partner.py

```diff
@@ -7,10 +7,10 @@
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
-    @api.onchange("zip_id")
-    def _onchange_zip_id(self):
-        res = super()._onchange_zip_id()
-        if self.zip_id and self.country_id.code == "TH":
-            address = self.zip_id.city_id.name.split(", ")
-            self.update({"street2": address[0], "city": address[1]})
-        return res
+    @api.depends("zip_id")
```

Comment:
You changed this to compute. You will need to add param `compute="_compute_zip_id` and `readonly=False` to the related fields also, right?

Please correct me if I am wrong about the new compute field.

Reply:
In module `base_location_geonames_import` has change structure city fields to compute field.
then, We will change it to compute too.